.github/workflows/inspektor-gadget.yml

```diff
@@ -444,11 +444,31 @@ jobs:
   release:
     name: Release
     # level: 2
-    needs: [documentation-checks, lint, test-integration, test-integration-aro, test-local-gadget, check-readme-up-to-date]
+    needs: [documentation-checks, lint, test-integration, test-local-gadget, check-readme-up-to-date]
     runs-on: ubuntu-latest
     if: startsWith(github.ref, 'refs/tags/v')
     steps:
     - uses: actions/checkout@v3
+    - name: Set IMAGE_TAG
+      run: |
+        TMP1=${GITHUB_REF#*/}
+        TMP2=${TMP1#*/}
+        IMAGE_TAG=${TMP2//\//-}
+        if [ "$IMAGE_TAG" = "main" ]; then
+            IMAGE_TAG="latest"
+        fi
+        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
+    - name: Build release YAML
+      run: |
+        # Use echo of cat to avoid printing a new line between files.
+        echo "$(cat pkg/resources/manifests/deploy.yaml) $(cat pkg/resources/crd/bases/gadget.kinvolk.io_traces.yaml)" > inspektor-gadget.yaml
+        export IMAGE="${{ env.REGISTRY }}/${{ env.RELEASE_CONTAINER_REPO }}:${IMAGE_TAG}"
+        perl -pi -e 's@(image:) ".+\"@$1 "$ENV{IMAGE}"@; s@"latest"@"$ENV{IMAGE_TAG}"@;' inspektor-gadget.yaml
+    - name: Add inspektor-gadget.yaml as artifact.
+      uses: actions/upload-artifact@master
+      with:
+        name: inspektor-gadget-yaml
+        path: /home/runner/work/inspektor-gadget/inspektor-gadget/inspektor-gadget.yaml
```

Comment:
I think this creates an artifact of the build but not an artifact of the release.

See the actions `csexton/release-asset-action@v2` and `actions/upload-release-asset@v1.0.1` below for the differences